source/NVDAObjects/UIA/wordDocument.py

```diff
@@ -334,17 +345,20 @@ def script_reportCurrentComment(self,gesture):
 			UIAElement=UIAElementArray.getElement(index)

```

Comment:
It really amazes me that this is not calling `getCommentInfoFromPosition`. Could you please change this in such a way that it uses that function and speaks a message based on what is in the dictionary? I think it is even possible to abstract this further and pas the result of CommentUIATextInfoQuickNavItem.label to ui.message.

If this sounds too complex, I"m happy to take this for you.

Reply:
It works with getCommentInfoFromPosition but when I tried CommentUIATextInfoQuickNavItem.label I got its adress in a memory or "Property object is not callable" error.

Reply:
I think it would be best to create a new _getPresentableCommentInfoFromPosition function that returns the label. Ten, you can make the label property return what that helper function returns for the particular textInfo, and same for the script.

Does that clear things up for you?

Reply:
I hope I understood you well. If so, should this new function has its doc string?
I also added a message for no comments in script.